js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_6
+ * by Diane Zevenbergen
+ */
+
+describe('Enhanced Date', function () {
+    var setDate = enhancedDate.setDate;
+    var getDate = enhancedDate.getDate;
+    var getDayName = enhancedDate.getDayName;
+    var getMonthName = enhancedDate.getMonthName;
+    var isToday = enhancedDate.isToday;
+    var isFuture = enhancedDate.isFuture;
+        
+    // setDate tests
+    describe('setDate method', function () {
+
+        it('should exist', function () {
+            expect(setDate()).to.exist;
+        });
+
+        it('should be instanceOf Date', function () {
+            expect(setDate()).is.an.instanceOf(Date);
+        });
+        
+        // test .deep.equal with 5/14/2015
+        it('matching dates should be deep equal', function () {
+            expect(setDate(new Date(2015,4,14))).to.deep.equal(new Date('5/14/2015'));
+        });
+
+        // test that .deep.equal fails comparing 5/14/2015 to current day
+        it('different dates should not be deep equal', function () {
+            expect(setDate(new Date(2015,4,14))).to.not.deep.equal(new Date());
+        });
+    });
+    
+    // getDate tests
+    describe('getDate method', function () {
+        
+        it('should exist', function () {
+            expect(getDate()).to.exist;
+        });
+        
+        // test that passing 'true' returns date obj
+        it('passing "true" should return instanceOf Date', function () {
+            expect(getDate(true)).is.an.instanceOf(Date);
+        });
+        
+        // test that passing nothing returns number (Epoch ms)
+        it('passing nothing should return a number', function () {
+            expect(getDate()).to.be.a('Number');
+        });
+        
+        // QUESTION:  is something like this too goofy for testing?
+        // test that getTime of returned date obj equals returned Epoch ms
+        it('getDate(true).getTime() should equal getDate()', function () {
+            expect(getDate(true).getTime()).to.equal(getDate());
+        });
+    });
+
+    // getDayName tests
+    describe('getDayName method', function () {
+        
+        it('should return a string', function () {
+            setDate();
+            expect(getDayName()).to.be.a('String');
+        });
+        
+        // set date to Thursday 5/14/15    
+        it('date = 5/14/15 should return "Thursday"', function () {
+            setDate(new Date(2015,4,14));
+            expect(getDayName()).to.equal('Thursday');
+        });
+
+        // set erroneous date
+        it('erroneous date should return undefined', function () {
+            setDate(new Date('nonsense'));
```

Comment:
This would be implementation specific. If setDate was designed to simply ignore the provided argument if it wasn't a Date object or a millisecond value, then this test would be invalid. That said, as long as it matches how you have documented this API it's what ultimately matters.